pwndbg/dbg/lldb/repl/__init__.py

```diff
@@ -1134,10 +1148,20 @@ def process_connect(driver: ProcessDriver, relay: EventRelay, args: List[str], d
         return
 
     io_driver = get_io_driver()
-    error = driver.connect(auto.target, io_driver, args.remoteurl, "gdb-remote")
-
-    if not error.success:
-        print_error(f"could not connect to remote process: {error.description}")
+    result = driver.connect(auto.target, io_driver, args.remoteurl, "gdb-remote")
+
+    error = False
+    match result:
+        case LaunchResultError(what, disconnected):
+            print_error(f"could not connect to remote: {what.description}")
+            if disconnected:
+                print_warn("disconnected")
+            error = True
+        case LaunchResultEarlyExit():
+            print_warn("remote exited early")
+            error = True
+
+    if error:
```

Comment:
Lines 1044-1054 have very similar logic, but they do `auto.close(); return` within the match cases instead.

The two differs in the error messages... shall we maybe put that in a helper function and just call it with proper err msgs?

Or just have it do the same logic here and there but consistently?

Reply:
I've changed them so they're the same now.